### DISCUSSION TITLE
Separate horizontal and vertical sensitivity

### DISCUSSION THREAD

**Discussion by okirema** [ppy/osu#17107] [Ideas]
Is it possible to add separate X and Y sens adjustment for osu stable? There could be a toggle button that says "Unlock separate sensitivity" or something similar. This is to give mouse players more customizability to match that of tablet players.

**Comment by Theighlin** [comment, marked as answer]
This repository is strictly for lazer development only, so this can only be taken as a proposal for the new client. As to whether is makes sense to have this or not, i can't really say.

**Reply by okirema** [reply]
I wasn't sure if github was the right place to ask about stable to be honest, so apologies for that.

As to whether it will make sense or not, this could be a way to improve the QoL for mouse players. Tablet players have their tablet drivers to change what they want, but mouse players don't have a similar counterpart.

**Reply by peppy** [reply]
We wouldn't be adding this for stable, and likely won't in lazer unless there is a fair amount of support for it.

**Reply by okirema** [reply]
Fair enough, thank you.

**Reply by IIIBlueberry** [reply]
> We wouldn't be adding this for stable, and likely won't in lazer unless there is a fair amount of support for it.

This is actually very needed feature for player who migrating to new aspect ratio and resolution due to how 2d mouse movement work with mouse CPI. with changing to same aspect ratio you can just multiply your sensitivity with your old res : new res to get your equivalent sensitivity. 

But for someone like me who also changing from 1680x1050 to 1920x1080 you don't a way to set separate equivalent horizontal and vertical sensitivity. I have to either overshoot vertically or undershoot horizontally.

**Comment by N-R-K** [comment]
> and likely won't in lazer unless there is a fair amount of support for it.

I'd like to see this implemented.

For some rationale: I play with a relatively high sensitivity (with mouse) and mainly aim with my wrist. For horizontal moment this works fine since the wrist has a pretty decent range of motion (ROM):

![image](https://github.com/user-attachments/assets/d7c2911a-5e16-4160-8a62-c7ca38890bbe)

However vertical movements need to be done with fingers. And moving the mouse up/down inside your palm, you have *very little* range of motion comparatively. (The mouse size matters here, smaller mice have slightly higher ROM, but it's still no where near the horizontal ROM you get from your wrist).

For this reason, I like to play with my y sens set higher than my x sens. IMO it makes sense to support this in-game.

**Reply by shinmorisawa** [reply]
i totally agree

**Reply by dotremilie** [reply]
I agree that it'd be nice to have it implemented.

>However vertical movements need to be done with fingers.

It doesn't necessarily _need_ to be done with fingers - I can serve as an example, because I don't move my fingers at all while playing.

**Comment by Aksumxd** [comment]
I was about to start a new thread on this, but I guess I'll just paste this here:

TLDR: I play with a 1.25x vertical sens, 1x horizontal sens. I have to install separate software to make this work (RawAccel on windows, YeetMouse on linux). If I'm playing on a friends computer, or at a lan party, or wherever else, it would be much nicer to just be able to change an option in the settings instead of having to install a program. 

This is not something most mouse players have historically changed, but I don't think its necessarily because they wouldn't like it. Tablet players have been playing with weird aspect ratios for ages, because that option is exposed to them through the tablet drivers. Sure, a 16:9 aspect ratio tablet area with 1:1 mapping to the screen is objectively "correct", but that doesn't mean its necessarily the most comfortable for everyone, and the amount of top players playing with weirdly shaped areas is proof of that. Why would that same thing not apply to mouse players?
 
I play with a low sensitivity and use my arm to aim large movements, and I have found that a horizontal movement pivoting from the elbow is much easier to do VS a vertical movement, where I have to move my whole arm up and down. A higher vertical sens makes everything feel way more comfortable for me, and it actually makes my aim feel more "balanced" (equal amount of "effort" for both horizontal and vertical movements). I think there could be others who feel the same if that option was given to them.

Also I feel like this wouldn't be too hard to implement into the game either.

**Reply by MattM0nkey** [reply]
I agree.

**Comment by peppy** [comment]
I've unmarked the answer here so this thread can be used to gather support for this potentially being implemented in the future.
